DOCSP-35982: Count usage example (#2850)
jenssegers/laravel-mongodb: docs/usage-examples/count.txt

.. _laravel-count-usage:

===============
Count Documents
===============

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: total, code example

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can count the number of documents returned by a query by calling the ``where()`` and

> ``count()`` methods on an object collection or a query builder.

Reviewer: Think Laravel name the document instance "model".

```suggestion
``count()`` methods on a collection of models, or a query builder.
```

To return the number of documents that match a filter, pass the query filter to the ``where()``
method and call the ``count()`` method.

Example
-------

This usage example performs the following actions:

- Uses the ``Movie`` Eloquent model to represent the ``movies`` collection in the
  ``sample_mflix`` database
- Counts the documents from the ``movies`` collection that match a query filter
- Prints the matching document count

The example calls the following methods on the ``Movie`` model:

> - ``where()``: matches documents in which the value of the ``genres`` field includes ``"Biography"``.
> - ``count()``: counts the number of matching documents. This method returns an integer value.

Reviewer 2: [Style Guide](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#colons:~:text=In%20a%20list%20item%2C%20if%20you%20need%20to%20separate%20an%20initial%20term%20or%20phrase%20from%20the%20information%20that%20follows%20it%2C%20use%20a%20colon.%20For%20example%3A)

```suggestion
- ``where()``: Matches documents in which the value of the ``genres`` field includes ``"Biography"``.
- ``count()``: Counts the number of matching documents. This method returns an integer value.
```
